source/opt/instruction.h

```diff
@@ -363,6 +363,7 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   // logical addressing rules when using logical addressing.  Normal validation
   // rules for physical addressing.
   bool IsValidBasePointer() const;
+  bool IsValidBaseImage() const;
```

Comment:
Would need a comment saying this can be used as the base pointer an OpImageRead (or OpImageWrite too??)

Reply:
Added a comment let me know if it is okay.